### PR TITLE
create_unsigned_tx accepts UTXOs

### PR DESCRIPTION
**Pull request by wizardofozzie**

### CODE REVIEW

**Comment by wizardofozzie** [comment]
adapted from my [pybitcointools fork](https://github.com/wizardofozzie/pybitcointools/blob/master/bitcoin/bci.py#L869-L910)

**Comment by mflaxman** [comment]
Great idea @wizardofozzie, thanks!

I'm requiring test coverage for all new PRs on the library, would you mind adding a unit test? Easy to copy example here (I threw out the private key for that address, so that UTXO will always remain unspent for testing):
https://github.com/blockcypher/blockcypher-python/blob/c84e896c9971a0c11f39f5235839f84621303393/test_blockcypher.py#L68-L87

If writing tests is a pain, I'm happy to do it for you.

File: blockcypher/api.py [hunk 1]
```diff
@@ -1499,6 +1500,15 @@ def create_unsigned_tx(inputs, outputs, change_address=None,
             inputs_cleaned.append({
                 'addresses': [address, ],
                 })
+        elif 'prev_hash' and  'output_index' in input_obj:
+            prev_hash = input_obj['prev_hash']
+            output_index = input_obj['output_index']
+            assert uses_only_hash_chars(prev_hash), prev_hash
+            assert str(output_index).isdigit(), output_index
```
Comment (mflaxman) [inline blockcypher/api.py]: Clever construction, I hadn't seen this before.

I don't love that we're converting an int to a string in order to check that it's a non-negative int.

I think `assert type(output_index) is int, output_index` is much clearer but two immediate problems come to mind:
- For python2 it would be possible for data to be represented as a `Long`, so we'd have to check for that as well. Yuck.
- This doesn't check the case of `output_index < 0` (your check does). Not the end of the world as the API should catch that anyway and return an error.

tl:dr; Lot of rambling. I don't have a better suggestion than what you've already done!

Comment (wizardofozzie) [inline blockcypher/api.py]: @mflaxman re unit test, yes, no problem. I'm using iOS (Pythonista), so it'll probably be tomorrow
1. My code uses input parameters of format `txid:index`: I find this to be clearest since outputs are also written as `address:value`. 
2. Perhaps [these regexs](https://github.com/wizardofozzie/pybitcointools/blob/master/bitcoin/pyspecials.py#L12-L27) would be worth considering to assert pubkey format

**Comment by wizardofozzie** [comment]
@mflaxman This seems to work, as I said, I'll need to test it on the laptop later tonight:

```
class CreateUnsignedTX(unittest.TestCase):

    def test_create_basic_unsigned(self):
        # This utxo correspons to a lost private key for 14MDR...
        create_unsigned_tx(
                inputs=[
                    {'prev_hash': 'ce25be65b41fb88cdd1bf4d799ad1b284822c86376f20e843c7c8d4ad2e4579f',
                     'output_index': 1,},
                    ],
                outputs=[
                    {
                        'value': 15500000,
                        'address': '14MDR5y8nHpqN7VtCKoDMMyfbeAD1HP9EW',
                        },
                    ],
                change_address=None,
                include_tosigntx=True,
                verify_tosigntx=True,
                coin_symbol='btc',
                api_key=BC_API_KEY,
                )
```

File: blockcypher/api.py [hunk 2]
```diff
@@ -1499,6 +1500,15 @@ def create_unsigned_tx(inputs, outputs, change_address=None,
             inputs_cleaned.append({
                 'addresses': [address, ],
                 })
+        elif 'prev_hash' and  'output_index' in input_obj:
+            prev_hash = input_obj['prev_hash']
```
Comment (wizardofozzie) [inline blockcypher/api.py]: Missing a `,`!

Comment (mflaxman) [inline blockcypher/api.py]: I don't see the missing `,`?

File: blockcypher/api.py (hunk 1 above)
Comment (mflaxman) [inline blockcypher/api.py]: Awesome regexes, thanks for the tip.

Will merge once test is in. Thanks again.

File: blockcypher/api.py [hunk 3]
```diff
@@ -1499,6 +1500,15 @@ def create_unsigned_tx(inputs, outputs, change_address=None,
...
+            output_index = input_obj['output_index']
+            assert uses_only_hash_chars(prev_hash), prev_hash
+            assert str(output_index).isdigit(), output_index
+            inputs_cleaned.append({
+                    'prev_hash': prev_hash
```
Comment (wizardofozzie) [inline blockcypher/api.py]: Missing `,` (corrected)

Comment (mflaxman) [inline blockcypher/api.py]: Good catch :+1: 

This is why the unit tests are now a must :)

File: blockcypher/api.py (hunk 2 above)
Comment (wizardofozzie) [inline blockcypher/api.py]: It was meant for line 1509
(Apologies, git and coding are never as easy using mobile devices)

File: blockcypher/api.py [hunk 4]
```diff
@@ -1499,6 +1500,15 @@ def create_unsigned_tx(inputs, outputs, change_address=None,
             inputs_cleaned.append({
                 'addresses': [address, ],
                 })
+        elif 'prev_hash' and  'output_index' in input_obj:
```
Comment (mflaxman) [inline blockcypher/api.py]: The first part of this will always return true. It should be:
```elif 'prev_hash' in input_obj and  'output_index' in input_obj:```

**Comment by mflaxman** [comment]
@wizardofozzie are you still planning on fixing this PR (and adding test coverage)? If not, I'm happy to do it for you.

**Comment by wizardofozzie** [comment]
@matthieu just reading this, I'll get on it now!

**Comment by quentinlesceller** [comment]
Closing for https://github.com/blockcypher/blockcypher-python/pull/75.
